llama_stack/providers/utils/inference/stores/sqlite.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+import json
+import os
+
+import aiosqlite
+
+from llama_stack.apis.inference import (
+    ListOpenAIChatCompletionResponse,
+    OpenAIChatCompletion,
+    OpenAICompletionWithInputMessages,
+    OpenAIMessageParam,
+    Order,
+)
+
+from ..inference_store import InferenceStore
+
+
+class SqliteInferenceStore(InferenceStore):
+    def __init__(self, conn_string: str):
+        self.conn_string = conn_string
+
+    async def initialize(self):
+        """Create the necessary tables if they don't exist."""
+        # Create directory if it doesn't exist
+        os.makedirs(os.path.dirname(self.conn_string), exist_ok=True)
+
+        async with aiosqlite.connect(self.conn_string) as conn:
```

Comment:
It seems like most of the code in this file already exists and is duplicated from other vector db implementations. Is there any chance to reuse some of those instead of adding another instance?

At a quick glance, only the table name would vary.